core/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TasksIT.java

```diff
@@ -30,9 +39,52 @@
 @ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.SUITE)
 public class TasksIT extends ESIntegTestCase {
 
-    public void testTaskCounts() {
+    public void testTaskCounts() throws Exception {
+        CountDownLatch latch = new CountDownLatch(1);
+        AtomicReference<ListTasksResponse> response = new AtomicReference<>();
+        final Client client;
+        boolean useDataClient = randomBoolean();
```

Comment:
if useDataClient is false you still might end up with a data client, right? its more like forceDataNodeClient?